aspnet/mvc/models/validation/sample/MoviesController.cs

```diff
@@ -0,0 +1,130 @@
+using System.Linq;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Data.Entity;
+using MVCMovie.Models;
+
+namespace MVCMovie.Controllers
+{
+    public class MoviesController : Controller
+    {
+        private MVCMovieContext _context;
+
+        public MoviesController(MVCMovieContext context)
+        {
+            _context = context;    
+        }
+
+        // GET: Movies1
+        public IActionResult Index()
+        {
+            return View(_context.Movie.ToList());
+        }
+
+        // GET: Movies1/Details/5
+        public IActionResult Details(int? id)
+        {
+            if (id == null)
+            {
+                return HttpNotFound();
+            }
+
+            Movie movie = _context.Movie.Single(m => m.Id == id);
```

Comment:
Use `SingleOrDefault()` or the next line won't be reached when the `Id` isn't matched.